tests/ui/features/lib/TextEditorPage.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/**
+ * ownCloud
+ *
+ * @author Phillip Davis <phil@jankaritech.com>
+ * @copyright 2017 Phillip Davis phil@jankaritech.com
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Page;
+
+use Behat\Mink\Session;
+use SensioLabs\Behat\PageObjectExtension\PageObject\Exception\ElementNotFoundException;
+use WebDriver\Key;
+
+/**
+ * Text Editor page.
+ */
+class TextEditorPage extends FilesPage {
+	protected $newTextFileButtonXpath
+		= './/div[contains(@class, "newFileMenu")]' .
+			'//a[@data-templatename="New text file.txt"]';
+	protected $newTextFileNameInputLabel = 'New text file.txt';
+	protected $newTextFileNameInputXpath
+		= './/div[contains(@class, "newFileMenu")]' .
+			'//a[@data-templatename="New text file.txt"]//input';
+	protected $textFileEditXpath = "//textarea[contains(@class,'ace_text-input')]";
+	protected $textEditorCloseButtonId = "editor_close";
+
+	/**
+	 * type in the field that matches the given xpath and press enter.
+	 * Note: this depends on methods that might only be in the Selenium
+	 * implementation
+	 *
+	 * @param string $xpath
+	 * @param string $text
+	 * @param Session $session
+	 * @throws \SensioLabs\Behat\PageObjectExtension\PageObject\Exception\ElementNotFoundException
+	 * @return void
+	 */
+	public function typeInFieldAndPressEnter($xpath, $text, Session $session) {
+		$element
+			= $session->getDriver()->getWebDriverSession()->element("xpath", $xpath);
```

Comment:
why is this new line happening?

Reply:
Coding standard said the line was to long.
It made me put the "=" on the next line.
Somehow it thinks this is shorter.

Reply:
yes by 1 character :-)

Reply:
that should also work:
```
$session->getDriver()
              ->getWebDriverSession()->element("xpath", $xpath);
```

```
$session->getDriver()->getWebDriverSession()->element(
    "xpath", $xpath
);
```

Reply:
I think phpcs is a bit stupid about the tabs (i.e. different to how my IDE visually displays them) so on my IDE the tabbed line goes past its 80 column dotted line. But counting tabs as 1, 2 or 4 spaces means the line fits.